ShellPkg/Application/Shell/Shell.c

```diff
@@ -2607,11 +2609,19 @@ RunCommandOrFile (
         CommandWithPath = ShellFindFilePathEx (FirstParameter, mExecutableExtensions);
       }
 
-      //
-      // This should be impossible now.
-      //
-      ASSERT (CommandWithPath != NULL);
+      // MU_CHANGE [START] - CodeQL change
+      if (CommandWithPath == NULL) {
+        //
+        // This should be impossible now.
+        //
+        ASSERT (CommandWithPath != NULL);
+        ShellPrintHiiEx (-1, -1, NULL, STRING_TOKEN (STR_SHELL_NOT_FOUND), ShellInfoObject.HiiHandle, FirstParameter);
+        SetLastError (SHELL_NOT_FOUND);
+        SHELL_FREE_NON_NULL (CommandWithPath);
+        return EFI_NOT_FOUND;

```

Comment:
nit: this PR has a mix of `return value;` and `return (value);`. I would lean towards the first, though I do see these files use the second some amount. At any rate, it would be nice to be consistent in the style, but no big deal to ignore it.